src/App.tsx

```diff
@@ -17,18 +75,28 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                applyStatusQuery={handleApplyStatusQuery}
+                applyTitleQuery={handleApplyTitleQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && <Loader />}
+              {!isLoading && (
+                <TodoList
+                  todos={filteredTodos}
+                  handleModalClick={handleModalClick}
+                  selectedTodoId={selectedTodo?.id || null}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
-
-      <TodoModal />
+      {selectedTodo !== null && (
+        <TodoModal todo={selectedTodo} handleCloseModal={handleCloseModal} />
```

Comment:
[checklist](https://github.com/mate-academy/react_dynamic-list-of-todos/blob/master/checklist.md): follow [these](https://medium.com/javascript-in-plain-english/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364) naming conventions for methods